src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -101,13 +104,18 @@ public synchronized void checkAndCreateLeasesForNewShards(IKinesisProxy kinesisP
      * @throws KinesisClientLibIOException
      */
     // CHECKSTYLE:OFF CyclomaticComplexity
-    private synchronized void syncShardLeases(IKinesisProxy kinesisProxy,
+    private synchronized void syncShardLeases(List<Shard> allShards, IKinesisProxy kinesisProxy,
             ILeaseManager<KinesisClientLease> leaseManager, InitialPositionInStreamExtended initialPosition,
             boolean cleanupLeasesOfCompletedShards, boolean ignoreUnexpectedChildShards)
             throws DependencyException, InvalidStateException, ProvisionedThroughputException,
             KinesisClientLibIOException {
-        List<Shard> shards = getShardList(kinesisProxy);
-        LOG.debug("Num shards: " + shards.size());
+        List<Shard> shards;
+        if(CollectionUtils.isNullOrEmpty(allShards)) {
+            shards = getShardList((kinesisProxy));
+        } else {
+            shards = allShards;
+        }
+        LOG.debug("Start shard syncing... The total number of shards found: " + shards.size());
```

Comment:
nit. avoid the dots in `Start shard syncing...`
The log will already have the class name. Sticking to "Num Shards:" should still be fine.

Reply:
Addressed